py/torch_tensorrt/dynamo/conversion/impl/pool.py

```diff
@@ -104,3 +108,184 @@ def max_poolNd(
 
     set_layer_name(pool_layer, target, name, source_ir)
     return pool_layer.get_output(0)
+
+
+def adaptive_avg_poolNd(
+    ctx: ConversionContext,
+    target: Union[Target, str],
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    output_size: Sequence[int],
+) -> TRTTensor:
+    input_rank = len(input.shape)
+
+    if input_rank == 3:  # TRT doesn't support 3D pooling
```

Comment:
This should also be part of a validator

Reply:
I think this should be fine. The native TRT doesn't support 3D pooling, as I commented, so I manually reshape to 4D. Torch-TRT can handle this case without throwing any error.